### PR TITLE
Add multiple components

### PR DESCRIPTION
**Pull request by tamarstern**
## Proposed Changes

  -
  -
  -

---
This change is [Reviewable](https://reviewable.io/reviews/teambit/bit/1210)

### CODE REVIEW

**Comment by tamarstern** [comment]
Add programatic api to add method for multiple components

File: e2e/commands/add-multiple.e2e.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  after(() => {
+    helper.destroyEnv();
+  });
+  describe('should add multiple components', function () {
+    it('should add multiple components that depends with each other and has spec files', function () {
+      this.timeout(10000);
+      helper.reInitLocalScope();
+      helper.copyFixtureComponents('add-multiple');
+      helper.npmLink('bit-bin');
+      helper.npmRunFile('add_multiple_test_files/index_three_components_with_specs.js');
```
Comment (GiladShoham) [inline e2e/commands/add-multiple.e2e.js]: get the output from the command, parse it and validate

File: e2e/fixtures/components/add-multiple/add_multiple_test_files/index_three_components_with_specs.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        }, 
+        {
+            'paths' : ['add_multiple_test_files/b.js'],
+            'main' : 'add_multiple_test_files/b.js'
+        }
+    ]
+}
+
+bitBin.addMany(components).then(function () {
+    console.log('success');
```
Comment (GiladShoham) [inline e2e/fixtures/components/add-multiple/add_multiple_test_files/index_three_components_with_specs.js]: log the response

File: src/api/consumer/lib/add.js [hunk 3]
```diff
@@ -1,12 +1,52 @@
...
-import type { AddProps, AddActionResults } from '../../../consumer/component-ops/add-components/add-components';
+import type {
+  AddProps,
+  AddActionResults,
+  PathOrDSL
+} from '../../../consumer/component-ops/add-components/add-components';
 import { loadConsumer, Consumer } from '../../../consumer';
 
-export default (async function addAction(addProps: AddProps): Promise<AddActionResults> {
+export async function addAction(addProps: AddProps): Promise<AddActionResults> {
```
Comment (GiladShoham) [inline src/api/consumer/lib/add.js]: change to addOne

File: e2e/commands/add-many.e2e.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  });
+  this.timeout(10000);
+  let nodeStartOutputObj;
+  describe('should add many components', function () {
+    before(() => {
+      helper.reInitLocalScope();
+      helper.copyFixtureComponents('add-many');
+      helper.linkNpm('bit-bin');
+      const nodeStartOutput = helper.nodeStart('add_many_test_files/index_four_components.js');
+      nodeStartOutputObj = JSON.parse(nodeStartOutput);
```
Comment (GiladShoham) [inline e2e/commands/add-many.e2e.js]: move status here

File: e2e/commands/add-many.e2e.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    });
+    it('should add 1 component no id and no spec', function () {
```
Comment (GiladShoham) [inline e2e/commands/add-many.e2e.js]: add case for component with multiple files

File: e2e/commands/add-many.e2e.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      expect(nodeStartOutputObj[0]).to.have.property('addedComponents');
+      expect(nodeStartOutputObj[0].addedComponents[0]).to.have.property('id');
+      expect(nodeStartOutputObj[0].addedComponents[0].id).to.equal('add_many_test_files/c');
+      expect(nodeStartOutputObj[0].addedComponents[0]).to.have.property('files');
+      expect(nodeStartOutputObj[0].addedComponents[0].files).to.be.array();
+      expect(nodeStartOutputObj[0].addedComponents[0].files).to.be.ofSize(1);
+      expect(nodeStartOutputObj[0].addedComponents[0].files[0].test).to.equal(false);
+      const status = helper.status();
+      expect(status).to.have.string('add_many_test_files/c ... ok');
+      const compData = helper.showComponent('add_many_test_files/c');
```
Comment (GiladShoham) [inline e2e/commands/add-many.e2e.js]: use show --json

**Pull request merged**